x-pack/plugins/observability_solution/logs_data_access/public/services/log_sources_service/index.ts

```diff
@@ -6,23 +6,24 @@
  */
 
 import { OBSERVABILITY_LOGS_DATA_ACCESS_LOG_SOURCES_ID } from '@kbn/management-settings-ids';
-import { LogSource } from '../../../common/types';
+import { LogSource, LogSourcesService } from '../../../common/services/log_sources_service/types';
 import { RegisterServicesParams } from '../register_services';
 
-export function createLogSourcesService(params: RegisterServicesParams) {
+export function createLogSourcesService(params: RegisterServicesParams): LogSourcesService {
   const { uiSettings } = params.deps;
   return {
-    getLogSources: (): LogSource[] => {
+    getLogSources: async () => {
```

Comment:
> [!NOTE]
> Do we need this function to be asynchronous? If the process is synchronous, it will make consuming the values in React easier as it won't require an async resolution with side effects.

Reply:
It's a good question. Whilst I'm not normally a fan of planning for things that haven't yet happened, as part of the original issues and addition of these services it was agreed to try and plan for a future where there are more log sources (not just the advanced setting), I can see a future where that will almost certainly involve async queries.

It does make the consumer use slightly more complex, but in my opinion doing this up front is better than the refactor effort later if we go from synchronous to asynchronous since many consumers / solutions will want to read the log sources over time.